Doc restructuring Plugin Install using core Libraries and with Node.js

pmario opened the pull request:
Restructuring Information about Plugin installation using core libraries and with client-server configurations using Node.js

This ZIP file can be used to test the changes: Start with the "Plugin" tiddler. 

TEST ME -->> [PR-restructuring-plugin-install.zip](https://github.com/Jermolene/TiddlyWiki5/files/8893243/PR-restructuring-plugin-install.zip)

The PR did remove many WikiLinks, to reduce the "noise" and replaced them with those links really needed. 

The Docs to install core plugins with the library is more interactive now as shown in the screenshot below. 

- Step 1: The description tiddler can be opened as a window. 
- Step 2: So the description can be read side by side with the wiki itself.
- Step 3: Eg: Clicking the ControlPanel link will open the ControlPanel tiddler in the main story river. 

![image](https://user-images.githubusercontent.com/374655/173411095-01862019-d7e9-4025-b01a-f98e1dfd2431.png)

-----------

The Plugins tiddler looks as follows now. It should have a much clearer structure.

![image](https://user-images.githubusercontent.com/374655/173411486-101c25a2-fa4f-402a-86c5-c88d64d298ec.png)

----------

Installing Plugins for Node.js should provide the links and the info needed in a more structural manner. With this info it should be possible to create some videos as requested at [Talk-TiddlyWiki](https://talk.tiddlywiki.org/t/migration-from-single-html-file-to-node-js/3585/7?u=pmario)

![image](https://user-images.githubusercontent.com/374655/173411729-f1573892-9dfb-4044-9c09-8757a5b61edd.png)

pmario commented:
.

Telumire commented:
The side by side windows is a good idea ! It's a bit off topic but what software did you use to position the window (the popup that appeared when you clicked on the maximize icon) ? 

![image](https://user-images.githubusercontent.com/31185220/173600977-d0ebfcea-bb75-4010-9fb6-a451987cd5c7.png)

Is that a windows 11 feature ?

pmario commented:
>Is that a windows 11 feature ?

Yes, but it is **on hover**, so _not_ click. .... For earlier versions there is a MicroSoft software named: PowerToys, which offers a similar functionality if you SHIFT drag a window

pmario commented:
@Jermolene ... It would be nice, if you could review this PR soon. I wanted to create some videos with it. With the PR merged it will probably be possible to create a smooth workflow in the videos. 

The main reason for the PR was this post at Talk: https://talk.tiddlywiki.org/t/migration-from-single-html-file-to-node-js/3585/7?u=pmario, which requested some more info about working with plugins using node.js. 

I think there is still more room for improvement, since information linked at:  https://tiddlywiki.com/#TiddlyWiki%20on%20Node.js also isn't structured in a way, that is easy to understand. ...

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing a plugin - client server.tid:
```diff
@@ -0,0 +1,38 @@
+created: 20220611123344385
+modified: 20220612091241651
+tags: [[TiddlyWiki on Node.js]] PluginsCS
+title: Installing a plugin - client server
```
This title isn't consistent with others. We hyphenate client-server, so it would be clearer to use the title "Installing official plugins on Node.js"

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing a plugin - client server.tid:
````diff
@@ -0,0 +1,38 @@
+}
+```
+
+<$macrocall $name=".note" _="An overview, how to handle plugins can be found at: [[Plugins]].<br>Also see: [[Installing custom plugins on Node.js]]."/>
````
The comma here is not needed in English. Clearer to say "An overview of working with plugins can be found at [[Plugins]]<br>Also see: [[Installing custom plugins on Node.js]]."/>`

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing a plugin - client server.tid:
```diff
@@ -0,0 +1,38 @@
+
+# Edit the `tiddlywiki.info` file (it is in JSON format) and locate the `plugins`, `themes` or `languages` section (see below)
+
+# Add entries corresponding to the plugins you wish to add. 
```
Our house style is not to use punctuation at the end of bullet list items.

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing a plugin - client server.tid:
```diff
@@ -0,0 +1,38 @@
+tags: [[TiddlyWiki on Node.js]] PluginsCS
+title: Installing a plugin - client server
+
+Follow these instructions when using TiddlyWiki under client-server (Node.js) configuration:
```
We need to be careful about nomenclature here. The client-server configuration is one of the configurations supported by the Node.js edition of TW5. The wording here gives the impression that Node.js is a synonym for client-server.

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing custom plugins on Node.js.tid:
```diff
@@ -1,27 +1,20 @@
 
+! Introduction
+
+There are several ways in which core and custom plugins can be installed when using TiddlyWiki with a Node.js client-server configuration.
```
I try to avoid the term "core plugins" because of the confusion from the fact that the core is itself a plugin. Instead I try to use "plugins from the official plugin library".

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Installing custom plugins on Node.js.tid:
```diff
@@ -1,27 +1,20 @@
+There are several ways in which core and custom plugins can be installed when using TiddlyWiki with a Node.js client-server configuration.
+
+
+<$macrocall $name=".note" _="""For instructions on installing plugins from libraries to ''single-file'' wikis head over to: [[Plugins]].
```
I am not keen on wording like "head over to". It's not helpful to use colloquialisms in technical documentation.

Jermolene commented on editions/tw5.com/tiddlers/nodejs/TiddlyWiki on Node.js.tid:
```diff
@@ -1,15 +1,20 @@
 title: TiddlyWiki on Node.js
 type: text/vnd.tiddlywiki
 
+\rules except wikilink
```
Why are you adding this rule? It's not consistent with other documentation tiddlers.

pmario replied:
I think topics like this need a bit more structure, to be understandable for new users. Especially TiddlyWiki and other CamelCase links adds a lot of noise. I tried to reduce those links to 1 appearance near the end of the tiddler.

Jermolene commented on editions/tw5.com/tiddlers/nodejs/TiddlyWiki on Node.js.tid:
```diff
@@ -1,15 +1,20 @@
+
 * The ability to build multiple wikis that blend different combinations of shared and unique content
 
+* You can edit your content on any suitably compatible HTML5 browser, including smartphones and tablets
```
Perhaps better to say "any modern browser". HTML5 isn't really a contemporary term.

Jermolene commented on editions/tw5.com/tiddlers/nodejs/Using TiddlyWiki on Node.js.tid:
```diff
@@ -1,18 +1,18 @@
 type: text/vnd.tiddlywiki
 
-TiddlyWiki5 includes a set of [[Commands]] for use on the command line to perform an extensive set of operations based on TiddlyWikiFolders, TiddlerFiles and TiddlyWikiFiles.
+~TiddlyWiki5 includes a set of commands for use on the command line to perform an extensive set of operations based on TiddlyWikiFolders, TiddlerFiles.
```
Why remove the link to commands? better to replace it with `[[commands|Commands]]`

pmario replied:
If users follow the link that early, imo it is a distruction. They didn't even read the basics about how to use the CLI. There is a specific link at the end of the tiddler. IMO that's the right place to follow a link to get more info

Jermolene commented on editions/tw5.com/tiddlers/plugins/Manually installing a plugin.tid:
```diff
@@ -1,14 +1,30 @@
 type: text/vnd.tiddlywiki
 
+\rules except wikilink
+\import Plugins
```
What is this import pragma intended to do? It doesn't look like there are any definitions in the tiddler "Plugins"

pmario replied:
There are some macro definitions in the Plugins tiddler. The `<<open-edition-window` macro used in the following line is defined there. ... I thought they are very specific. BUT at a second look I think I can make them generic. So they will be available for the whole doc, to create a "side by side" view.

Jermolene commented on editions/tw5.com/tiddlers/plugins/Manually installing a plugin.tid:
```diff
@@ -1,14 +1,30 @@
+
+# The plugin should now be available for use
+
+<$macrocall $name=".note" _="If you want to install a plugin using the core plugin library, have a closer look at: [[Installing a plugin from the plugin library]]<br>An overview can be found at: [[Plugins]]"/>
```
Again, this is clumsily worded, and not consistent with other docs. Why does the plugins link get preceded with a colon? We try to write hypertextually, linking words in the flow of a sentence.

I don't like "Have a closer look"; again it's too colloquial.

Jermolene commented on editions/tw5.com/tiddlers/plugins/Plugin Ordering.tid:
```diff
@@ -0,0 +1,51 @@
+tags: PluginMechanism
+title: Plugin Ordering
+
+Using the Node.js client-server configuration, plugins are activated in the following order:
```
The comma here is not needed in English. "With" would be better than "Until".

Jermolene commented on editions/tw5.com/tiddlers/plugins/Plugin Ordering.tid:
```diff
@@ -0,0 +1,51 @@
+
+''Important:''
+
+<<<
```
It looks like you're using the quote syntax for emphasis, and not for an actual quote. It would be better to avoid that, and reserve quotes for their intended semantic purpose.

Jermolene commented on editions/tw5.com/tiddlers/plugins/Plugin Ordering.tid:
```diff
@@ -0,0 +1,51 @@
+<<<
+''Elements lower in the list take precedence''.
+
+Including a plugin as an ordinary tiddler by ''drag & drop''  into the browser, will result in the plugin only being active in the browser. It will not be available under Node.js!
```
Drag and drop shouldn't be in bold; and we should spell "and" in full. It's important to be consistent with the existing docs.

Also the exlamation mark is not necessary here.

Jermolene commented on editions/tw5.com/tiddlers/plugins/Plugin Ordering.tid:
```diff
@@ -0,0 +1,51 @@
+Option 2:
+
+<<<
+Option 2 is used, if you want to work with a fixed configuration. Updates have to be performed independently.
```
The phrase "performed independently" is not clear; What does it mean?

Jermolene commented on editions/tw5.com/tiddlers/plugins/Plugin Ordering.tid:
```diff
@@ -0,0 +1,51 @@
+Option 3:
+
+<<<
+This option allows you ''add'' and start an edition, without the need to overwrite an existing `tiddlywiki.info` file. It allows you to start any edition as a client-server edition, even if the tiddlywiki.info file didn't specify it.
```
The first comma isn't needed.

Jermolene reviewed (commented):
Thanks @pmario. It took me a LONG time to do this review because there are a lot of divergences from the establish docs style. If you've got a lot of changes coming up it would be helpful to make smaller PRs.

pmario commented:
@Jermolene ... I did fix the "wording problems". I hope I got them all. 

I'm sorry, that so many files have been changed but it had to be done in one go since the existing structure didn't work out, if a user was searching for `Plugins`. With this PR the search results and the tiddler interconnection should be better. 



1. The doc macro definitions have been moved to `$:/editions/tw5.com/wikitext-macros` and renamed, so they should be generic enough to be re-used more often, if needed. They are extremely handy to create a "side by side" view as shown in the screenshot and the video
  1.1. The `\import Plugins` pragma has been removed
2. The blockquotes in `Plugin Ordering` have been removed and replaced by ULs
3. The new `- client server` tiddlers have been renamed to `Installing official plugins on Node.js.tid` and `Uninstalling a plugin with Node.js.tid` as suggested
4. There is a new tag `PluginsCS`, which is used to create a sorted list in the `Plugins` tiddler. It defines a subset, that is needed to describe the plugin installation mechanism, without overwhelming new users with the list: [TiddlyWiki on Node.js](https://tiddlywiki.com/#TiddlyWiki%20on%20Node.js) shows. 
5. `Uninstalling a plugin with Node.js.tid` is new as described in 3.
6. `Installing a plugin from the plugin library` is more interactive now as shown in the first screenshot and the video
7. `Manually installing a plugin` same as 6.
8. `Plugin Ordering` is a new tiddler. ... It was completely missing from `PluginMechanism` tiddler. 
9. `Plugin Folder` should have a clearer structure now.
10. `Plugins` defines 2 sections: "Handling a single file wiki" and "Handling a client-server config with Node.js
   - This should be less confusing for new users
11. `Uninstalling a plugin` ... only describes the single-file wiki behaviour and is interactive using a side-by-side view

The ZIP contains the changed tiddlers for easy testing --> [pr-restructuring-plugin-install-01.zip](https://github.com/Jermolene/TiddlyWiki5/files/8928276/pr-restructuring-plugin-install-01.zip)
